src/bundler/esbuild_bundler.ts

```diff
@@ -0,0 +1,35 @@
+import { denoPlugins, esbuild } from "../deps.ts";
+
+export type EsbuildBundleOptions = {
+  /** The path to the file being bundled */
+  entrypoint: string;
+  /** The path to the deno.json / deno.jsonc config file. */
+  configPath: string;
+  /** specify the working directory to use for the build */
+  absWorkingDir: string;
+};
+
+export const EsbuildBundler = {
+  bundle: async (options: EsbuildBundleOptions): Promise<Uint8Array> => {
+    try {
+      Deno.version.deno;
+      // esbuild configuration options https://esbuild.github.io/api/#overview
+      const result = await esbuild.build({
+        entryPoints: [options.entrypoint],
+        platform: "neutral",
+        target: "deno1", // TODO: the versions should come from the user defined input
```

Comment:
Esbuild requires a deno version for the target value, currently the user is not able to specify the version their app should run on. Once they are able to, I believe the version they specify should be used here